integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/session/SearchSessionFlushIT.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.session;
+
+import static org.hibernate.search.util.impl.integrationtest.orm.OrmUtils.withinTransaction;
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.FlushMode;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.GenericField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.work.spi.PojoWorkPlan;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.integrationtest.orm.OrmSetupHelper;
+import org.hibernate.search.util.impl.test.annotation.TestForIssue;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Tests the behaviour of different kinds of {@link Session#flush()}.
+ * Each of ones has to trigger a {@link PojoWorkPlan#prepare()} exactly when the flush is expected.
+ */
+@TestForIssue( jiraKey = "HSEARCH-3360" )
+public class SearchSessionFlushIT {
```

Comment:
Same here, I think you should move this to `AutomaticIndexingBasicIT` or move this class to the same package.